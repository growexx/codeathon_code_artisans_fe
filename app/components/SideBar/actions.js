import {
  ADD_SIDEBAR_ITEM,
  LOAD_SIDEBAR_ITEMS,
  LOAD_SIDEBAR_ITEMS_SUCCESS,
  REMOVE_SIDEBAR_ITEM,
} from './constants';

export const addSidebarItem = data => ({
  type: ADD_SIDEBAR_ITEM,
  data,
});

export const removeSidebarItem = id => ({
  type: REMOVE_SIDEBAR_ITEM,
  id,
});

export const loadSidebarItems = () => ({
  type: LOAD_SIDEBAR_ITEMS,
});

export const loadDataSuccess = data => ({
  type: LOAD_SIDEBAR_ITEMS_SUCCESS,
  data,
});
