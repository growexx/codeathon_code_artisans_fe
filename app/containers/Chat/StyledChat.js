import styled from 'styled-components';

export const StyledChat = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  background-color: @chat-bot-bg;

  .chat-section-wrapper {
    overflow-y: auto;
    padding-bottom: 50px;
  }

  .input-section-wrapper {
    width: 50vw;
    margin: 30px auto;
    margin-top: 10px;
    @media only screen and (max-width: 1000px) {
      width: 90%;
    }
  }

  .chat-input {
    color: white;
    background-color: rgb(0, 0, 0);
    border-color: rgba(32, 33, 35, 0.5);
    border-radius: 0.75rem;
    padding: 16px;
    padding-right: 64px;
  }

  .chat-input:focus {
    box-shadow: none;
  }

  .default-option-container {
    display: grid;
    gap: 0.75rem;
    grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
    grid-auto-flow: row;
    width: 100%;
    padding: 0rem 0.5rem;
  }

  .default-options {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 10px;
    gap: 0.75rem;
  }

  .default-options .option-card {
    font-weight: 600;
    width: 100%;
    height: 100%;
    text-align: left;
    overflow: hidden;
    color: rgb(197, 197, 210);
    background-color: #091b29;
    padding: 8px 12px;
    margin: 5px 10px;
    border: 1px solid rgb(86, 88, 105);
    border-radius: 0.75rem;
    text-overflow: ellipsis;
    cursor: pointer;
  }

  .default-options .option-card .card-content {
    font-weight: 500;
    opacity: 0.5;
    text-overflow: ellipsis;
    overflow: hidden;
  }

  .option-card:hover {
    cursor: pointer;
    background-color: #102a3f;
  }

  .send-btn {
    postition: absolute;
    bottom: 44px;
    left: calc(50vw - 60px);
    background-color: @primary-color;
    border: none;
    color: white;
    padding: 0px 14px;

    @media only screen and (max-width: 1000px) {
      left: calc(100% - 60px);
    }
  }

  @media only screen and (max-width: 576px) {
    .default-options {
      flex-direction: column;
    }

    .default-options.options-1 {
      display: none;
    }
  }
`;
