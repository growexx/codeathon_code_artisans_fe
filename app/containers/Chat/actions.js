import {
  GET_CHAT_HISTORY,
  ADD_CHAT_QUESTION,
  SET_CHAT_HISTORY,
  ADD_CHAT_ANSWER,
} from './constants';

export const getChatHistory = () => ({
  type: GET_CHAT_HISTORY,
});

export const addChatQuestion = (chatId, question) => ({
  type: ADD_CHAT_QUESTION,
  chatId,
  question,
});

export const addChatAnswer = (chatId, answer, sources) => ({
  type: ADD_CHAT_ANSWER,
  chatId,
  answer,
  sources,
});

export const setChatHistory = (chatId, history) => ({
  type: SET_CHAT_HISTORY,
  chatId,
  history,
});
