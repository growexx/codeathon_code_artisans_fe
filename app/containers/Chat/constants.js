export const GET_CHAT_HISTORY = 'llm/Chat/GET_CHAT_HISTORY';
export const ADD_CHAT_QUESTION = 'llm/Chat/ADD_CHAT_QUESTION';
export const ADD_CHAT_ANSWER = 'llm/Chat/ADD_CHAT_ANSWER';
export const SET_CHAT_HISTORY = 'llm/Chat/SET_CHAT_HISTORY';

export const suggesstions = [
  {
    title:
      "Which product segments or business units contributed the most to the Clean Science's revenue and profitability?",
    titlePart1: 'Which product segments or business units',
    titlePart2:
      "contributed the most to the Clean Science's revenue and profitability?",
  },
  {
    title:
      "What are the major cost drivers affecting the Galaxy Surfactants's profitability, and how have they been managed?",
    titlePart1: 'What are major cost drivers affecting',
    titlePart2:
      "Galaxy Surfactants' profitability and how have they been managed?",
  },
  {
    title:
      'What are the overall financial performance highlights of Gujarat Fluorochemicals in the last fiscal year?',
    titlePart1: 'What are the overall financial',
    titlePart2:
      'performance highlights of Gujarat Fluorochemicals in the last fiscal year?',
  },
  {
    title:
      'What are the key risks and challenges the EPL faces in the chemical industry, and how are they being addressed?',
    titlePart1: 'What are the key risks and challenges EPL',
    titlePart2: 'faces in chemical industry and how are they being addressed?',
  },
];
