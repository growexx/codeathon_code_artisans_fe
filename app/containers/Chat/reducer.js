/* eslint-disable no-case-declarations */
/* eslint-disable consistent-return */
/* eslint-disable no-param-reassign */
import produce from 'immer';
import {
  ADD_CHAT_ANSWER,
  ADD_CHAT_QUESTION,
  SET_CHAT_HISTORY,
} from './constants';

export const initialState = {
  chatHistory: {},
};

const sidebarReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case ADD_CHAT_QUESTION:
        if (draft.chatHistory[action.chatId]) {
          draft.chatHistory[action.chatId].push({
            question: action.question,
          });
        } else {
          draft.chatHistory[action.chatId] = [
            {
              question: action.question,
            },
          ];
        }
        break;
      case ADD_CHAT_ANSWER:
        const { length } = draft.chatHistory[action.chatId];
        draft.chatHistory[action.chatId][length - 1] = {
          ...draft.chatHistory[action.chatId][length - 1],
          answer: action.answer,
          sources: action.sources,
        };
        break;
      case SET_CHAT_HISTORY:
        draft.chatHistory[action.chatId] = action.history;
        break;
      default:
        return state;
    }
  });

export default sidebarReducer;
