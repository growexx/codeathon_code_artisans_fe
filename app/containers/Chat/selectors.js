import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectData = state => state.chat || initialState;

const selectChatHistory = () =>
  createSelector(
    selectData,
    dataState => dataState.chatHistory,
  );

export { selectChatHistory, selectData };
