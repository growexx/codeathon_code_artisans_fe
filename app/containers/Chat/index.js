import React, { useState, useEffect, useRef } from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import PropTypes from 'prop-types';
import { Link, useLocation, useNavigate, useParams } from 'react-router-dom';
import {
  Button,
  Input,
  Skeleton,
  notification,
  Modal,
  Upload,
  Form,
} from 'antd';
import { SendOutlined, UploadOutlined, PlusOutlined } from '@ant-design/icons';
import { useInjectReducer } from 'utils/injectReducer';
import request from 'utils/request';
import { StyledChat } from './StyledChat';
import ChatItem from '../../components/ChatItem';
import reducer from './reducer';
import { selectChatHistory } from './selectors';
import { addChatAnswer, addChatQuestion, setChatHistory } from './actions';
import { addSidebarItem } from '../../components/SideBar/actions';
import { suggesstions } from './constants';
import { API_ENDPOINTS, ROUTES } from '../constants';

const key = 'chat';
const tempAns =
  'Lorem ipsum dolor sit amet consectetur adipisicing elit. Esse quod dolore, accusantium harum eaque natus veniam facilis. Odio ipsam accusamus fugiat natus omnis illum unde quae corrupti amet est. Iusto!';

const Chat = ({
  isNew,
  chatHistory,
  addChatQue,
  setHistory,
  addChatAns,
  addNewSidebar,
}) => {
  useInjectReducer({ key, reducer });

  const { chatId } = useParams();
  const [searchValue, setSearchValue] = useState('');
  const [loading, setLoading] = useState(false);
  const [typing, setTyping] = useState(false);
  const [firstRender, setFirstRender] = useState(true);
  const [userScrolling, setUserScrolling] = useState(false);
  const chatSectionRef = useRef(null);
  const location = useLocation();
  const navigate = useNavigate();
  const pathname = location.pathname.split('/')[1];
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [uploadForm] = Form.useForm();

  const handleSubmit = e => {
    if (e.key === 'Enter' && !e.shiftKey) {
      e.preventDefault();
      handleChatMessage(searchValue);
    }
  };

  const handleChatMessage = async message => {
    if (message.trim() === '') {
      return;
    }
    setLoading(true);
    setUserScrolling(false);
    try {
      if (firstRender && isNew) {
        window.history.pushState(null, '', `/chat/${chatId}`);
        addNewSidebar({ chatId, name: message.trim() });
        setFirstRender(false);
      }
      addChatQue(chatId, message.trim());
      setSearchValue('');
      // const response = await request(API_ENDPOINTS.CHAT, {
      //   method: 'POST',
      //   body: { key: chatId, question: message.trim() },
      // });
      const response = {
        status: 1,
        data: {
          answer: tempAns,
        },
      };
      if (response.status === 1) {
        const answer = response.data.answer;
        const sources = getSources(response.data.sources);
        addChatAns(chatId, answer, sources);
      }
    } catch (error) {
      notification.error({ message: error.message });
    }
    setTyping(true);
  };

  const loadChatHistory = async () => {
    setLoading(true);
    try {
      if (chatHistory[chatId] === undefined) {
        // const response = await request(`${API_ENDPOINTS.CHAT}?id=${chatId}`, {
        //   method: 'GET',
        // });
        const response = {
          data: {
            question: 'This is a test question',
            answer: tempAns,
            chat_history: {
              question: 'This is a test question',
              answer: tempAns,
            },
          },
        };
        const chatHistory = [];
        if (response?.status === 1) {
          chatHistory.push({
            question: response?.data?.question,
            answer: response?.data?.answer,
            sources: getSources(response?.data?.sources),
          });
          response?.data?.chat_history?.map((chat, index) => {
            chatHistory.push({
              question: chat.question,
              answer: chat.answer,
              sources: getSources(chat.sources),
            });
          });
        } else {
          navigate(ROUTES.HOME);
        }
        setHistory(chatId, chatHistory);
      }
    } catch (error) {
      notification.error({ message: error.message });
    }
    setLoading(false);
  };

  const shouldShowTyping = index => {
    if (typing && index === chatHistory[chatId].length - 1) {
      return true;
    }
    return false;
  };

  const scrollToBottom = () => {
    const element = document.getElementById('chat');
    if (!userScrolling) {
      element.scrollTop = element.scrollHeight;
    }
  };

  const handleScroll = () => {
    const element = document.getElementById('chat');
    const isScrolledToBottom =
      element.scrollHeight - element.scrollTop === element.clientHeight;
    setUserScrolling(!isScrolledToBottom);
  };

  const handleUploadModal = () => {
    setIsModalOpen(true);
  };

  const handleOk = () => {
    console.log(uploadForm.getFieldsValue());
    const data = new FormData();
    const uploadValue = uploadForm.getFieldValue('upload');
    console.log(uploadValue);
    uploadValue.forEach(x => {
      if (x.originFileObj !== undefined) {
        data.append('pdf', x.originFileObj);
      }
    });
    console.log(data);
    setIsModalOpen(false);
  };
  const handleCancel = () => {
    setIsModalOpen(false);
  };

  useEffect(() => {
    if (!isNew) {
      loadChatHistory();
    }
    document.getElementById('chat').addEventListener('scroll', handleScroll);

    return () => {
      document
        .getElementById('chat')
        ?.removeEventListener('scroll', handleScroll);
      setTyping(false);
      setFirstRender(true);
      setSearchValue('');
      setLoading(false);
    };
  }, [chatId]);

  return (
    <StyledChat>
      <div className="chat-section-wrapper" id="chat" ref={chatSectionRef}>
        {chatHistory[chatId] !== undefined || pathname === 'new-chat'
          ? chatHistory[chatId]?.map((chat, index) => (
              <div key={`${chatId}${index.toString()}`}>
                <ChatItem content={chat.question} />
                {chat?.answer ? (
                  <ChatItem
                    content={chat.answer}
                    bot
                    typing={shouldShowTyping(index)}
                    scrollToBottom={scrollToBottom}
                    setLoading={setLoading}
                  />
                ) : (
                  <ChatItem
                    skeleton
                    content={<Skeleton active />}
                    bot
                    scrollToBottom={scrollToBottom}
                  />
                )}
                {scrollToBottom()}
              </div>
            ))
          : [1].map((chat, index) => (
              <div key={`index${index.toString()}`}>
                <ChatItem
                  skeleton
                  content={<Skeleton active title paragraph={false} />}
                />
                <ChatItem skeleton content={<Skeleton active />} bot />
              </div>
            ))}
      </div>
      <div className="input-section-wrapper">
        {isNew && firstRender && (
          <div className="default-option-container">
            <div className="default-options options-1">
              {suggesstions.map((suggestion, index) => {
                if (index < 2) {
                  return (
                    <Button
                      key={`option-btn-${index.toString()}`}
                      data-testid="SUGG_BTN"
                      className="option-card"
                      onClick={() => {
                        handleChatMessage(suggestion.title);
                      }}
                    >
                      <div>{suggestion.titlePart1}</div>
                      <div className="card-content">
                        {suggestion.titlePart2}
                      </div>
                    </Button>
                  );
                }
              })}
            </div>
            <div className="default-options">
              {suggesstions.map((suggestion, index) => {
                if (index > 1) {
                  return (
                    <Button
                      key={`option-btn-${index.toString()}`}
                      data-testid="SUGG_BTN"
                      className="option-card"
                      onClick={() => {
                        handleChatMessage(suggestion.title);
                      }}
                    >
                      <div>{suggestion.titlePart1}</div>
                      <div className="card-content">
                        {suggestion.titlePart2}
                      </div>
                    </Button>
                  );
                }
              })}
            </div>
          </div>
        )}
        <div className="input-wrapper">
          <Button className="upload-btn" onClick={() => handleUploadModal()}>
            <UploadOutlined />
          </Button>
          <Input.TextArea
            onPressEnter={handleSubmit}
            autoSize={{ minRows: 1, maxRows: 8 }}
            placeholder="Send a message"
            className="chat-input"
            value={searchValue}
            disabled={loading}
            onChange={e => setSearchValue(e.target.value)}
          />
          {
            <Button
              className="send-btn"
              data-testid="SEND_CHAT_BTN"
              onClick={() => handleChatMessage(searchValue)}
            >
              <SendOutlined />
            </Button>
          }
        </div>
      </div>
      <Modal
        title="Basic Modal"
        open={isModalOpen}
        onOk={handleOk}
        onCancel={handleCancel}
      >
        <Form role="form" form={uploadForm}>
          <Form.Item
            label="Upload pdf"
            valuePropName="fileList"
            name="upload"
            getValueFromEvent={uploadedFiles =>
              uploadedFiles && uploadedFiles.fileList
            }
          >
            <Upload
              beforeUpload={() => false}
              listType="picture-card"
              data-testid="upload"
            >
              <div>
                <PlusOutlined />

                <div
                  style={{
                    marginTop: 8,
                  }}
                >
                  Upload
                </div>
              </div>
            </Upload>
          </Form.Item>
        </Form>
      </Modal>
    </StyledChat>
  );
};

Chat.propTypes = {
  chatHistory: PropTypes.object,
  addChatQue: PropTypes.func,
  setHistory: PropTypes.func,
  addNewSidebar: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  chatHistory: selectChatHistory(),
});

export function mapDispatchToProps(dispatch) {
  return {
    addChatQue: (chatId, question) =>
      dispatch(addChatQuestion(chatId, question)),
    addChatAns: (chatId, answer, sources) =>
      dispatch(addChatAnswer(chatId, answer, sources)),
    setHistory: (chatId, history) => dispatch(setChatHistory(chatId, history)),
    addNewSidebar: data => dispatch(addSidebarItem(data)),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(Chat);
