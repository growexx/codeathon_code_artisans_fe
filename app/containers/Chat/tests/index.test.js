import React from 'react';
import { useParams } from 'react-router-dom';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import history from 'utils/history';
import request from 'utils/request';
import { HistoryRouter as Router } from 'redux-first-history/rr6';
import Chat, { getSources, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';
import { setChatHistory } from '../actions';
import 'jest-dom/extend-expect';

jest.mock('components/ChatItem');
jest.mock('utils/request');
jest.mock('react-router-dom', () => ({
  useParams: jest.fn(() => ({ chatId: '1' })),
  useLocation: jest.fn(() => ({ pathname: 'chat/1' })),
  useNavigate: jest.fn(() => () => {}),
}));

let globalStore;
const props = {
  isNew: false,
};
const componentWrapper = Component =>
  render(
    <Provider store={globalStore}>
      <IntlProvider locale="en">
        <Router history={history}>
          <Component {...props} />
        </Router>
      </IntlProvider>
    </Provider>,
  );
describe('<Chat />', () => {
  beforeAll(() => {
    const { store } = configureStore({});
    globalStore = store;
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper(Chat);
    expect(firstChild).toMatchSnapshot();
  });
  it('get sources function', () => {
    expect(getSources(['abc/test.pdf', 'xyz/test.pdf'])).toStrictEqual([
      'test.pdf',
    ]);
  });
  it('mapDispatch to props', () => {
    const mockFn = jest.fn();
    const eventObject = {
      target: {
        value: 'test',
      },
      preventDefault: jest.fn(),
    };
    const returnValue = mapDispatchToProps(mockFn);
    returnValue.addChatQue(eventObject);
    returnValue.addChatAns(eventObject);
    returnValue.setHistory(eventObject);
    returnValue.addNewSidebar(eventObject);
    expect(mockFn).toBeCalled();
  });
  it('type question and submit by enter key', async () => {
    useParams.mockResolvedValueOnce(undefined);
    request.mockResolvedValueOnce({
      status: 1,
      data: {
        question: 'question',
        answer: 'answer',
        sources: ['123.pdf'],
        chat_history: [
          { question: 'question', answer: 'answer', sources: ['123.pdf'] },
        ],
      },
    });
    const { getByPlaceholderText } = componentWrapper(Chat);
    await waitFor(() => {
      fireEvent.change(getByPlaceholderText('Send a message'), {
        target: { value: '' },
      });
      fireEvent.keyDown(getByPlaceholderText('Send a message'), {
        key: 'Enter',
        keyCode: 13,
      });
    });
  });
  it('type question and not submit by shift + enter key', async () => {
    useParams.mockResolvedValueOnce(1);
    request.mockResolvedValueOnce({
      status: 0,
    });
    const { getByPlaceholderText } = componentWrapper(Chat);
    await waitFor(() => {
      fireEvent.change(getByPlaceholderText('Send a message'), {
        target: { value: '' },
      });
      fireEvent.keyDown(getByPlaceholderText('Send a message'), {
        key: 'shift',
        keyCode: 16,
      });
      fireEvent.keyDown(getByPlaceholderText('Send a message'), {
        key: 'enter',
        keyCode: 13,
      });
    });
  });
  it('type question and submit by button click', async () => {
    useParams.mockResolvedValueOnce(1);
    const { getByPlaceholderText, getByTestId } = componentWrapper(Chat);
    await waitFor(() => {
      fireEvent.change(getByPlaceholderText('Send a message'), {
        target: { value: '' },
      });
      fireEvent.click(getByTestId('SEND_CHAT_BTN'));
    });
  });
  it('ask question and show answer', async () => {
    useParams.mockResolvedValueOnce(1);
    const { getByPlaceholderText, getByTestId } = componentWrapper(Chat);
    await waitFor(() => {
      fireEvent.change(getByPlaceholderText('Send a message'), {
        target: { value: '' },
      });
      fireEvent.click(getByTestId('SEND_CHAT_BTN'));
    });
  });
  it('show chat history', () => {
    document.getElementById = jest.fn(() => ({
      scrollHeight: 0,
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    }));
    globalStore.dispatch(
      setChatHistory(1, [
        { question: 'question', answer: 'answer', sources: [] },
      ]),
    );
    componentWrapper(Chat);
  });
  it('new chat', async () => {
    props.isNew = true;
    request.mockResolvedValueOnce({
      status: 1,
      data: {
        question: 'question',
        answer: 'answer',
        sources: ['123.pdf'],
        chat_history: [
          { question: 'question', answer: 'answer', sources: ['123.pdf'] },
        ],
      },
    });
    request.mockResolvedValueOnce({
      status: 1,
      data: { answer: 'answer', sources: [] },
    });
    const { getAllByTestId, getByPlaceholderText } = componentWrapper(Chat);
    fireEvent.change(getByPlaceholderText('Send a message'), {
      target: { value: '' },
    });
    fireEvent.click(getAllByTestId('SUGG_BTN')[0]);
  });
  it('new chat and clicking suggestions', async () => {
    props.isNew = true;
    request.mockResolvedValueOnce({
      status: 1,
      data: {
        question: 'question',
        answer: 'answer',
        sources: ['123.pdf'],
        chat_history: [
          { question: 'question', answer: 'answer', sources: ['123.pdf'] },
        ],
      },
    });
    request.mockResolvedValueOnce({
      status: 1,
      data: { answer: 'answer', sources: [] },
    });
    const { getAllByTestId, getByPlaceholderText } = componentWrapper(Chat);
    fireEvent.change(getByPlaceholderText('Send a message'), {
      target: { value: 'a' },
    });
    fireEvent.click(getAllByTestId('SUGG_BTN')[3]);
  });
});
