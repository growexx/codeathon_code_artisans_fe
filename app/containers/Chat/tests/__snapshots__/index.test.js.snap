// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> Should render and match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
  background-color: #091b29;
}

.c0 .chat-section-wrapper {
  overflow-y: auto;
  padding-bottom: 50px;
}

.c0 .input-section-wrapper {
  width: 50vw;
  margin: 30px auto;
  margin-top: 10px;
}

.c0 .chat-input {
  color: white;
  background-color: #000000;
  border-color: rgba(32,33,35,0.5);
  border-radius: 0.75rem;
  padding: 16px;
  padding-right: 64px;
}

.c0 .chat-input:focus {
  box-shadow: none;
}

.c0 .default-option-container {
  display: grid;
  gap: 0.75rem;
  grid-template-columns: repeat(auto-fit,minmax(250px,1fr));
  grid-auto-flow: row;
  width: 100%;
  padding: 0rem 0.5rem;
}

.c0 .default-options {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 10px;
  gap: 0.75rem;
}

.c0 .default-options .option-card {
  font-weight: 600;
  width: 100%;
  height: 100%;
  text-align: left;
  overflow: hidden;
  color: #c5c5d2;
  background-color: #091b29;
  padding: 8px 12px;
  margin: 5px 10px;
  border: 1px solid #565869;
  border-radius: 0.75rem;
  text-overflow: ellipsis;
  cursor: pointer;
}

.c0 .default-options .option-card .card-content {
  font-weight: 500;
  opacity: 0.5;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c0 .option-card:hover {
  cursor: pointer;
  background-color: #102a3f;
}

.c0 .send-btn {
  postition: absolute;
  bottom: 44px;
  left: calc(50vw - 60px);
  background-color: #ff6b6b;
  border: none;
  color: white;
  padding: 0px 14px;
}

@media only screen and (max-width:1000px) {
  .c0 .input-section-wrapper {
    width: 90%;
  }
}

@media only screen and (max-width:1000px) {
  .c0 .send-btn {
    left: calc(100% - 60px);
  }
}

@media only screen and (max-width:576px) {
  .c0 .default-options {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c0 .default-options.options-1 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="chat-section-wrapper"
    id="chat"
  >
    <div />
  </div>
  <div
    class="input-section-wrapper"
  >
    <div
      class="input-wrapper"
    >
      <textarea
        class="ant-input chat-input ant-input-disabled"
        disabled=""
        placeholder="Send a message"
        style="height: -256px; resize: none; min-height: -32px; max-height: -256px;"
      />
      <button
        class="ant-btn ant-btn-default send-btn"
        data-testid="SEND_CHAT_BTN"
        type="button"
      >
        <span
          aria-label="send"
          class="anticon anticon-send"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="send"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M931.4 498.9L94.9 79.5c-3.4-1.7-7.3-2.1-11-1.2a15.99 15.99 0 00-11.7 19.3l86.2 352.2c1.3 5.3 5.2 9.6 10.4 11.3l147.7 50.7-147.6 50.7c-5.2 1.8-9.1 6-10.3 11.3L72.2 926.5c-.9 3.7-.5 7.6 1.2 10.9 3.9 7.9 13.5 11.1 21.5 7.2l836.5-417c3.1-1.5 5.6-4.1 7.2-7.1 3.9-8 .7-17.6-7.2-21.6zM170.8 826.3l50.3-205.6 295.2-101.3c2.3-.8 4.2-2.6 5-5 1.4-4.2-.8-8.7-5-10.2L221.1 403 171 198.2l628 314.9-628.2 313.2z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
